build_tools/benchmarks/collect_compilation_statistics.py

```diff
@@ -130,6 +133,81 @@ def get_module_path(flag_file: TextIO) -> Optional[str]:
   return module_path
 
 
+def get_module_map_from_generation_config(
+    serialized_gen_config: TextIO, e2e_test_artifacts_dir: pathlib.PurePath
+) -> Dict[CompilationInfo, pathlib.Path]:
+  gen_configs = serialization.unpack_and_deserialize(
+      data=json.load(serialized_gen_config),
+      root_type=List[iree_definitions.ModuleGenerationConfig])
+  module_map = {}
+  for gen_config in gen_configs:
+    model = gen_config.imported_model.model
+    compile_config = gen_config.compile_config
+    target_archs = []
+    for compile_target in compile_config.compile_targets:
+      arch = compile_target.target_architecture
+      target_archs.append(
+          (f"{arch.type.value}-{arch.architecture}-{arch.microarchitecture}-"
+           f"{compile_target.target_abi.value}"))
+    compilation_info = CompilationInfo(
+        model_name=model.name,
+        model_tags=tuple(model.tags),
+        model_source=model.source_type.value,
+        target_arch=f"[{','.join(target_archs)}]",
+        compile_tags=tuple(compile_config.tags))
+    module_dir_path = iree_artifacts.get_module_dir_path(
+        module_generation_config=gen_config, root_path=e2e_test_artifacts_dir)
+    module_path = module_dir_path / iree_artifacts.MODULE_FILENAME
+    module_map[compilation_info] = pathlib.Path(module_path)
+
+  return module_map
+
+
+def get_module_map_from_benchmark_suite(
```

Comment:
This function is basically copied from line 169 in the original code.

Reply:
Thanks for flagging that :-)